File: src/main/java/io/github/jho951/assignment/job/web/dto/ImageJobResultResponse.java
package io.github.jho951.assignment.job.web.dto;

import java.time.Instant;

public record ImageJobResultResponse(
        String jobId,
        String status,
        String result,
        Instant completedAt,
        Instant expiresAt,
        ImageJobErrorResponse error
) {
}

File: src/main/kotlin/io/github/jho951/assignment/job/web/dto/ImageJobResultResponse.kt
package io.github.jho951.assignment.job.web.dto

import java.time.Instant
import kotlin.jvm.JvmRecord

@JvmRecord
data class ImageJobResultResponse(
    val jobId: String,
    val status: String,
    val result: String?,
    val completedAt: Instant?,
    val expiresAt: Instant?,
    val error: ImageJobErrorResponse?
)
